libs/vkd3d/swapchain.c

```diff
@@ -2421,23 +2416,25 @@ static HRESULT d3d12_swapchain_init(struct d3d12_swapchain *swapchain, IDXGIFact
         }
     }
 
-    IDXGIFactory_AddRef(swapchain->factory = factory);
-
-    if (FAILED(hr = d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
+    if (FAILED(d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
```

Comment:
`FAILED(hr = d3d12_swapchain_set_fullscreen`?

Reply:
It should, good catch.